pkg/testutils/pgtest/datadriven.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL.txt and at www.mariadb.com/bsl11.
+//
+// Change Date: 2022-10-01
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt and at
+// https://www.apache.org/licenses/LICENSE-2.0
+
+package pgtest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils/datadriven"
+	"github.com/jackc/pgx/pgproto3"
+)
+
+// Walk walks path for datadriven files and calls RunTest on them.
+func Walk(t *testing.T, path, addr, user string) {
+	datadriven.Walk(t, path, func(t *testing.T, path string) {
+		RunTest(t, path, addr, user)
+	})
+}
+
+// RunTest executes PGTest commands, connecting to the database specified by
```

Comment:
I'll chance a guess that you'd really enjoy https://github.com/cockroachdb/datadriven/pull/1 (once it's done, I'm improving the API still)